Add devstack settings file

dan-f opened the pull request:
@mulby @dsjen I added a basic devstacks settings file.

I'd prefer if it didn't have to import selectively from `local.py`, but I didn't want to import the `DATABASES` setting, for example.

Furthermore, I tried to factor out common settings between `local.py` and `devstack.py` into a `dev.py` file, but this caused numerous issues for me where settings were not getting applied as expected. I think it had to do with the order in which these module-local variables are attached to the `django.conf.settings` object when imported.

mulby commented on analytics_dashboard/settings/devstack.py:
```diff
@@ -0,0 +1,28 @@
+
+# Default oauth settings used by LMS devstack
+SOCIAL_AUTH_EDX_OIDC_KEY = 'YOUR_OAUTH2_KEY'
+SOCIAL_AUTH_EDX_OIDC_SECRET = 'secret'
```
Aren't these defined in production? Is non-DRY stuff you brought up in stand-up?

dan-f replied:
These are actually not defined in `production.py`.

The DRY stuff I mentioned is the selective importing of settings from `local.py`. Ideally we could do something like `from analytics_dashboard.settings.dev import *`.

mulby replied:
Odd that you need this... I'm seeing the settings in /edx/etc/insights.yml

Ideally they wouldn't be defined both in the config file and in the django settings.

dsjen replied:
That does seem a bit odd.  Is it that configurations in `insights.yml` aren't getting picked up?  Does anyone know how devops sets these?

dan-f replied:
They're manually parsed from `insights.yml` by `production.py`, but I can try scraping them here as well. I can't remember now why I re-declared them here :P

mulby replied:
I'd strongly prefer we only define these in the edx/configuration repo, keeping them in sync will be... tricky otherwise.

dsjen commented on analytics_dashboard/settings/devstack.py:
```diff
@@ -0,0 +1,28 @@
+SOCIAL_AUTH_EDX_OIDC_KEY = 'YOUR_OAUTH2_KEY'
+SOCIAL_AUTH_EDX_OIDC_SECRET = 'secret'
+
+DATA_API_URL = os.getenv('DATA_API_URL', DATA_API_URL)
```
Are you thinking that these should be passed in when starting up Insights?

dan-f replied:
Yes, this pattern allows the developer to test against different data APIs.

dsjen replied:
Ok, makes sense!

dan-f commented:
@dsjen @mulby I pushed up a commit which I think should have addressed most concerns. The one part I don't like is that `dev.py` has to import from `base.py` since it appends to `INSTALLED_APPS` and `MIDDLEWARE_CLASSES`.

mulby commented on analytics_dashboard/settings/dev.py:
```diff
@@ -0,0 +1,90 @@
+########## TOOLBAR CONFIGURATION
+# See: http://django-debug-toolbar.readthedocs.org/en/latest/installation.html#explicit-setup
+
+if os.environ.get('ENABLE_DJANGO_TOOLBAR', False):
```
I think you actually want: `if os.environ.get('ENABLE_DJANGO_TOOLBAR', 'false').lower() in ('true', 't', '1'):`

Note that `os.environ.get` will return a string if the environment variable is actually defined.

dan-f replied:
Nice. But are we really expecting any value other than 'True'?

mulby commented on analytics_dashboard/settings/dev.py:
```diff
@@ -0,0 +1,90 @@
+)
+TEST_RUNNER = 'django_nose.NoseTestSuiteRunner'
+
+LMS_COURSE_SHORTCUT_BASE_URL = 'https://courses.edx.org/courses'
```
Not sure what this is used for, but shouldn't we default it to something like "https://example.com/courses"?

dan-f replied:
Good point. I think I copied this over from `local.py`. It looks like it's used to construct "View Live" links.

mulby commented on analytics_dashboard/settings/dev.py:
```diff
@@ -0,0 +1,90 @@
+LMS_COURSE_SHORTCUT_BASE_URL = 'https://courses.edx.org/courses'
+
+########## BRANDING
+PLATFORM_NAME = 'edX'
```
I would make this default to something other than 'edX', maybe 'Open edX'?

mulby commented on analytics_dashboard/settings/devstack.py:
```diff
@@ -0,0 +1,12 @@
+
+
+if os.environ.get('ENABLE_DJANGO_TOOLBAR', False):
+    from analytics_dashboard.settings.local import DEBUG_TOOLBAR_PATCH_SETTINGS
```
Hrm... is this overridden by the yaml and we have to re-override it?

If so, I think I would prefer just explicitly redefining them here instead of importing from local. Making devstack depend on local just to get this feels too complex to me.

dan-f replied:
Embarassing answer: I don't know why this is here. Must be some cruft left over from the refactor, since it's not overridden by the yaml, and there's no need to re-override it. It indeed should inherit this behavior already from `dev.py`.

dan-f commented:
@mulby I think I handled your last round of feedback. Let me know how things look now :)

mulby commented:
:+1: once tests pass

dsjen commented on analytics_dashboard/settings/local.py:
```diff
@@ -38,74 +22,3 @@
-LMS_COURSE_SHORTCUT_BASE_URL = 'https://courses.edx.org/courses'
-
-########## BRANDING
-PLATFORM_NAME = 'edX'
```
Local acceptance tests fail at `test_error_pages.py` line 23 because the name is expected to be "edX Insights" and the name has now changed to be "Open edx Insights" rather.  I've no preference of whether we fix the test or move the configuration to local.

dsjen commented:
My only question is about the `PLATFORM_NAME` configuration.  Otherwise, LGTM!

dan-f commented:
Thanks @dsjen. I changed the acceptance setting to match `'Open edX'` in the spirit of openness. I also rebased for accessibility test fixes. Here's to hoping tests pass...

dsjen commented:
Congrats on the first merged PR to Insights! :rabbit: :zap: :rotating_light:

dan-f commented:
:bow:
